src/wp-includes/class-wp-scripts.php

```diff
@@ -1058,119 +899,106 @@ private function is_delayed_strategy( $strategy ) {
+
+		// If this handle isn't registered, don't filter anything and return.
 		if ( ! isset( $this->registered[ $handle ] ) ) {
-			return '';
+			return $eligible;
+		}
+
+		// If the handle is not enqueued, don't filter anything and return.
+		if ( ! $this->query( $handle, 'enqueued' ) ) {
+			return $eligible;
 		}
 
+		$is_alias = (bool) ! $this->registered[ $handle ]->src;
 		$intended_strategy = $this->get_data( $handle, 'strategy' );
 
-		/*
-		 * Handle known blocking strategy scenarios.
-		 * - An empty strategy is synonymous with blocking.
-		 * - A script alias (where $src is false) must always be blocking since the after inline script cannot be
-		 *   delayed as there is no external script tag and thus no load event at which the inline script can be run.
-		 */
-		if ( empty( $intended_strategy ) ) {
-			return '';
+		// For non-alias handles, an empty intended strategy filters all strategies.
+		if ( ! $is_alias && empty( $intended_strategy ) ) {
+			return array();
 		}
 
-		// Aliases that have before/after inline scripts can never be delayed since there is no load event.
-		if ( ! $this->registered[ $handle ]->src && $this->has_inline_script( $handle ) ) {
-			return '';
+		// Handles with inline scripts attached in the 'after' position cannot be delayed.
+		if ( $this->has_inline_script( $handle, 'after' ) ) {
+			return array();
 		}
 
-		// Handle async strategy scenario.
-		if ( 'async' === $intended_strategy && $this->has_only_delayed_dependents( $handle, true ) ) {
-			return 'async';
+		// If the intended strategy is 'defer', filter out 'async'.
+		if ( $intended_strategy === 'defer' ) {
+			$eligible =  array( 'defer' );
 		}
 
-		// Handling defer strategy scenarios. (The intended strategy could actually be 'async' here.)
-		if ( $this->has_only_delayed_dependents( $handle ) ) {
-			return 'defer';
+		$dependents = $this->get_dependents( $handle );
+
+		// Recursively filter eligible strategies for dependents.
+		foreach ( $dependents as $dependent ) {
+			// Bail early once we know the eligible strategy is blocking.
+			if ( empty( $eligible ) ) {
+				return array();
+			}
+
+			$eligible = $this->filter_eligible_strategies( $dependent, $eligible, $checked );
 		}
 
-		return '';
+		return $eligible;
 	}
 
 	/**
 	 * Gets data for inline scripts registered for a specific handle.
 	 *
 	 * @since 6.3.0
 	 *
-	 * @param string $handle Name of the script to get data for.
-	 *                       Must be lowercase.
+	 * @param string $handle   Name of the script to get data for. Must be lowercase.
+	 * @param string $position The position of the inline script.
+	 *
 	 * @return bool Whether the handle has an inline script (either before or after).
 	 */
-	private function has_inline_script( $handle ) {
+	private function has_inline_script( $handle, $position = null ) {
+		if ( $position && in_array( $position, array( 'before', 'after' ), true ) ) {
+			return $this->get_data( $handle, $position );
```

Comment:
Since per the docs this should return a boolean (which also makes sense given the function name), we have to cast it here to return the correct type.
```suggestion
			return (bool) $this->get_data( $handle, $position );
```

Reply:
Fixed in [12969ea](https://github.com/10up/wordpress-develop/pull/81/commits/12969ead675c93e2f1aa83bddf038e5f20ab96f0)